File: ConsoleApplication1/Text.cpp
#include "Text.h"
using namespace std;

void Text::addToText(vector<char> chars) {
	Line word = Line(chars);
	this->text.push_back(word);
}

void Text::getContent() {

	for (Line element : this->text) {
		element.getContest();
	}
}

void Text::delLine() {
	this->text.pop_back();
}

void Text::clearText() {
	this->text.clear();
}

int Text::getShortestLen() {
	int min = this->text[0].strLenght();
	for (Line element : this->text) {
		int l = element.strLenght();
		if (l < min) min = l;
	}
	return min;
}

float Text::percVoicr() {
	int voice = 0;
	int voiced = 0;
	for (Line element : this->text) {
		voice += element.voiceCount();
		voiced += element.voicedCount();
	}
	float perc = float(voiced) / (float(voice) + float(voiced)) * 100.;
	return perc;
}

void Text::delAllDS() {
	vector<Line> temp;
	for (Line newLine : this->text) {
		newLine.deleteDoubleSpaces();
		temp.push_back(newLine);
	}
	this->text.swap(temp);
}

void Text::delAllSES() {
	for (int index = 0; index < this->text.size(); index++) {
		text[index].delStartEndSpaces();
	}
}

File: Lab2 C#/Lab2/Text.cs
using System;
using Line_;

namespace Text_
{
    public class Text
    {
        private Line[] _text = new Line[0];

        public void addToText(char[] chars)
        {
            Array.Resize(ref _text, _text.Length + 1);
            Line word = new Line(chars);
            _text[^1] = word;
        }
        public void delLine()
        {
            Array.Resize(ref _text, _text.Length - 1);
        }
        public void ClearText() 
        {
            _text = new Line[0];
        }
        public int SmallestLength() { 
            int minimum = _text[0].lineLenght();
            foreach (var element in _text) // Check every string
            {
                if (element.lineLenght() < minimum)
                {
                    minimum = element.lineLenght();
                }
            }
            return minimum;
        }

        public float PercVoiced()
        {
            float voices = 0;
            float voiced = 0;
            foreach (var element in _text)
            {
                voices += element.VoicesAmount();
                voiced += element.VoicedAmount();
            }
            float perc = voiced / (voiced + voices) * 100;
            return perc;
        }
        public void GetContent()
        {
            foreach (var element in _text)
            {
                element.GetContest();
            }
            Console.WriteLine();
        }

        public void RemoveDoubleSpaces()
        {
            foreach (var element in _text)
            {
                element.RemoveRepeatSpaces();
            }
        }

        public void RemoveStartEndSpaces()
        {
            foreach (var element in _text)
            {
                element.RemoveStartEndSpaces();
            }
        }   

    }
}
